src/main/kotlin/no/nav/familie/ef/sak/iverksett/oppgaveforbarn/BarnFyllerÅrOppfølgingsoppgaveService.kt

```diff
@@ -59,115 +51,96 @@ class BarnFyllerÅrOppfølgingsoppgaveService(
-        val personerMedTermindatoBarn = barnTilUtplukkForOppgave.filter { it.termindatoBarn != null && it.fødselsnummerBarn == null }
-        val pdlPersonMedForelderBarnRelasjon = pdlClient.hentPersonForelderBarnRelasjon(personerMedTermindatoBarn.map { it.fødselsnummerSøker })
+        val barnUtenFødselsnummer =
+            barnTilUtplukkForOppgave.filter { it.termindatoBarn != null && it.fødselsnummerBarn == null }
+        return barnTilUtplukkForOppgave.filter { it.fødselsnummerBarn != null } +
+            finnFødselsnummerTilTerminbarn(barnUtenFødselsnummer)
+    }
 
-        val returnBarnTilUtplukkForOppgave = mutableListOf<BarnTilUtplukkForOppgave>()
+    private fun finnFødselsnummerTilTerminbarn(barnMedTermindato: List<BarnTilUtplukkForOppgave>): List<BarnTilUtplukkForOppgave> {
+        val pdlPersonMedForelderBarnRelasjon =
+            pdlClient.hentPersonForelderBarnRelasjon(barnMedTermindato.map { it.fødselsnummerSøker })
 
-        for (personMedTermindatoBarn in personerMedTermindatoBarn) {
-            val termindato = personMedTermindatoBarn.termindatoBarn!!
-            val pdlPersonMedForelderBarnRelasjonData = pdlPersonMedForelderBarnRelasjon[personMedTermindatoBarn.fødselsnummerSøker] ?: error("Finner ikke pdldata for søker=${personMedTermindatoBarn.fødselsnummerSøker}")
-            val forelderBarnRelasjoner = pdlPersonMedForelderBarnRelasjonData.forelderBarnRelasjon.mapNotNull { it.relatertPersonsIdent }
+        return barnMedTermindato.map { barn ->
+            val termindato = barn.termindatoBarn!!
+            val søkerPdlData = pdlPersonMedForelderBarnRelasjon[barn.fødselsnummerSøker]
+                ?: error("Finner ikke pdldata for søker=${barn.fødselsnummerSøker}")
+            val forelderBarnRelasjoner = søkerPdlData.forelderBarnRelasjon.mapNotNull { it.relatertPersonsIdent }
             val besteMatch = finnBesteMatchPåFødselsnummerForTermindato(forelderBarnRelasjoner, termindato)
 
-            returnBarnTilUtplukkForOppgave.add(personMedTermindatoBarn.copy(fødselsnummerBarn = besteMatch))
+            barn.copy(fødselsnummerBarn = besteMatch)
         }
-        return returnBarnTilUtplukkForOppgave + barnTilUtplukkForOppgave.filter { it.fødselsnummerBarn != null }
     }
 
-    private fun filtrerBarnSomHarFyltÅr(barnTilUtplukkForOppgave: List<BarnTilUtplukkForOppgave>): Set<OpprettOppgaveForBarn> {
-        val barnTilUtplukkMedTermindatoBarn = hentFødselsnummerTilTermindatoBarn(barnTilUtplukkForOppgave)
-        val barnPersonIdenter = barnTilUtplukkMedTermindatoBarn.mapNotNull { it.fødselsnummerBarn }
-
-        if (barnPersonIdenter.isEmpty()) return setOf()
+    private fun lagOpprettOppgaveForBarn(barnTilUtplukkForOppgave: List<BarnTilUtplukkForOppgave>): Set<OpprettOppgaveForBarn> {
+        val barn = hentFødselsnummerTilTermindatoBarn(barnTilUtplukkForOppgave)
```

Comment:
Denne er litt vrien - for den "henter" jo alle barn, men prøver å oppdatere terminbarn med fødselsnummer... 
`settFødselsnummerForTerminbarn` / ... 

Vanskelig å finne et godt funksjonsnavn her